api/controllers/datasets.py

```diff
@@ -48,6 +48,41 @@ def update(credentials, did):
     return util.json_encode({"success": "ok"})
 
 
+@bottle.delete("/datasets/delete/<did:int>")
+@_auth.requires_auth
+def delete(credentials, did):
+    dm = DatasetModel()
+    dataset = dm.get(did)
+    ensure_owner_or_admin(dataset.tid, credentials["id"])
+
+    tm = TaskModel()
+    task = tm.get(dataset.tid)
+
+    delta_metric_types = [
+        config["type"]
+        for config in ujson.loads(task.annotation_config_json)["delta_metrics"]
+    ]
+    delta_metric_types.append(None)
+
+    s3_client = boto3.client(
+        "s3",
+        aws_access_key_id=config["eval_aws_access_key_id"],
+        aws_secret_access_key=config["eval_aws_secret_access_key"],
+        region_name=config["eval_aws_region"],
+    )
+
+    for perturb_prefix in delta_metric_types:
+        s3_client.delete_object(
```

Comment:
We also need to delete the regular dataset from s3 too, where there is not a perturb prefix. Does your code do that?

Reply:
yep, on line 65 `None` is added to `delta_metric_types`, which will delete the regular dataset